src/test/EndToEnd1.java

```diff
@@ -0,0 +1,92 @@
+package test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import processing.IChanges;
+import processing.ProcessingFactory;
+import processing.Processor;
+import processing.utility.Pixel;
+
+/**
+* This file does an End to End test of the KoraKaagaz application. The test
+* is achieved by two clients connecting to the same board where one client 
+* acts as a sender and the other as a receiver. The sender client sends some 
+* pixels and the receiver client tries to catch the pixels. The sent pixels 
+* and the received pixels must match for the test to pass. 
+* @author Muhammed Yaseen
+*
+*/
+
+public class EndToEnd1 implements IChanges {
+    public static void main(String[] args) throws InterruptedException, IOException {
+        
+        ProcessBuilder processBuilder1 = new ProcessBuilder();
+        ProcessBuilder processBuilder2 = new ProcessBuilder();
+        
+        // Create Main Server
+        processBuilder1.command(
+            "java", 
+            "-jar", 
+            "MainServer.jar"
+        );
+        Process process1 = processBuilder1.start();
+        // Wait for main server to start
+        Thread.sleep(3000);
+        
+        // Create Board
+        Processor processor = ProcessingFactory.getProcessor();
+        // IP of main server verified by running MainServer.jar separately
+        String ip = "192.168.1.6";
```

Comment:
This is a dynamic IP address (changes from time to time and depends totally upon the DHCP on your local network). High chances that your system might have lost this IP (and received something else from the DHCP) and the test won't work on your system locally as well. Needless to say, it won't work on mine.

You need to figure out a way to get the IP dynamically to get these tests running.

Reply:
If there's no other better you find, I suggest keeping `IP` as a command line argument.

Reply:
You can use getIP of networking to get the IP of your machine. As main server and client will be running on same machine for this test it's fine.